core/primitives/src/block.rs

```diff
@@ -2,372 +2,33 @@ use std::cmp::max;
-            prev_state_root,
-            outcome_root,
-            timestamp,
-            next_bp_hash,
-            block_merkle_root,
-        );
-        let inner_rest = BlockHeaderInnerRest::new(
-            chunk_receipts_root,
-            chunk_headers_root,
-            chunk_tx_root,
-            chunks_included,
-            challenges_root,
-            random_value,
-            validator_proposals,
-            chunk_mask,
-            gas_price,
-            total_supply,
-            challenges_result,
-            last_final_block,
-            last_ds_final_block,
-            approvals,
-        );
-        let (hash, signature) = signer.sign_block_header_parts(prev_hash, &inner_lite, &inner_rest);
-        Self { prev_hash, inner_lite, inner_rest, signature, hash }
-    }
-
-    pub fn genesis(
-        height: BlockHeight,
-        state_root: MerkleHash,
-        chunk_receipts_root: MerkleHash,
-        chunk_headers_root: MerkleHash,
-        chunk_tx_root: MerkleHash,
-        chunks_included: u64,
-        challenges_root: MerkleHash,
-        timestamp: DateTime<Utc>,
-        initial_gas_price: Balance,
-        initial_total_supply: Balance,
-        next_bp_hash: CryptoHash,
-    ) -> Self {
-        let inner_lite = BlockHeaderInnerLite::new(
-            height,
-            EpochId::default(),
-            EpochId::default(),
-            state_root,
-            CryptoHash::default(),
-            to_timestamp(timestamp),
-            next_bp_hash,
-            CryptoHash::default(),
-        );
-        let inner_rest = BlockHeaderInnerRest::new(
-            chunk_receipts_root,
-            chunk_headers_root,
-            chunk_tx_root,
-            chunks_included,
-            challenges_root,
-            CryptoHash::default(),
-            vec![],
-            vec![],
-            initial_gas_price,
-            initial_total_supply,
-            vec![],
-            CryptoHash::default(),
-            CryptoHash::default(),
-            vec![],
-        );
-        let hash = BlockHeader::compute_hash(CryptoHash::default(), &inner_lite, &inner_rest);
-        Self {
-            prev_hash: CryptoHash::default(),
-            inner_lite,
-            inner_rest,
-            signature: Signature::empty(KeyType::ED25519),
-            hash,
-        }
-    }
-
-    pub fn hash(&self) -> CryptoHash {
-        self.hash
-    }
-
-    /// Verifies that given public key produced the block.
-    pub fn verify_block_producer(&self, public_key: &PublicKey) -> bool {
-        self.signature.verify(self.hash.as_ref(), public_key)
-    }
-
-    pub fn timestamp(&self) -> DateTime<Utc> {
-        from_timestamp(self.inner_lite.timestamp)
-    }
-
-    pub fn num_approvals(&self) -> u64 {
-        self.inner_rest.approvals.iter().filter(|x| x.is_some()).count() as u64
-    }
-}
-
 #[derive(BorshSerialize, BorshDeserialize, Serialize, Debug, Clone, Eq, PartialEq)]
-pub struct Block {
+pub struct BlockV1 {
```

Comment:
why `V1`?

Reply:
for future, when we add V2 to not rename.

Reply:
I guess my question was why 1 instead of `PROTOCOL_VERSION`, but it makes sense now that I think more about it.